homeassistant/components/gpsd/config_flow.py

```diff
@@ -38,7 +38,10 @@ async def async_step_user(
         if user_input is not None:
             self._async_abort_entries_match(user_input)
 
-            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock: socket.socket = await self.hass.async_add_executor_job(
+                lambda: socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            )
+
             try:
                 sock.connect((user_input[CONF_HOST], user_input[CONF_PORT]))
                 sock.shutdown(2)
```

Comment:
I am wondering, doesn't this connect also count as blocking IO?

Maybe just move all socket operations into a separate function and throw that in the executor

Reply:
Done, thanks :)